Microsoft.NET.Build.Containers.IntegrationTests/DockerSupportsArchInlineData.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using System.Text.Json;
+using Microsoft.DotNet.CommandUtils;
+using Xunit.Sdk;
+
+namespace Microsoft.NET.Build.Containers.IntegrationTests;
+public class DockerSupportsArchInlineData : DataAttribute
+{
+    private readonly string _arch;
+    private readonly object[] _data;
+
+    public DockerSupportsArchInlineData(string arch, params object[] data)
+    {
+        _arch = arch;
+        _data = data;
+    }
+
+    public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+    {
+
+        if (DaemonSupportsArch(_arch))
+        {
+            var totalArray = new object[_data.Length + 1];
+            totalArray[0] = _arch;
+            Array.Copy(_data, 0, totalArray, 1, _data.Length);
+            return new[] { totalArray };
+        };
+        return Array.Empty<object[]>();
+    }
+
+    private bool DaemonSupportsArch(string quemu_Arch)
+    {
+        if (new LocalDocker(m => { return; }).IsAvailable().GetAwaiter().GetResult())
```

Comment:
the blocking call second time - i guess it's the time for non-async version of the call.

Quickly checked: nothing blocking that, can be another improvement